apps/aechannel/src/aesc_fsm.erl

```diff
@@ -2716,27 +2957,40 @@ default_minimum_depth(responder) -> ?MINIMUM_DEPTH.
 
 start_min_depth_watcher(Type, SignedTx, Updates,
                         #data{watcher = Watcher0,
-                              opts = #{minimum_depth := MinDepth}} = D) ->
+                              opts = #{minimum_depth := MinDepth} = Opts} = D) ->
     Tx = aetx_sign:tx(SignedTx),
     TxHash = aetx_sign:hash(SignedTx),
     evt({tx_hash, TxHash}),
     Nonce = aetx:nonce(Tx),
     evt({nonce, Nonce}),
     {OnChainId, D1} = on_chain_id(D, Tx),
     case {Type, Watcher0} of
-        {funding, undefined} ->
+        {{?MIN_DEPTH, ?WATCH_FND = Sub}, undefined} ->
+            WOpts = maps:get(watcher, Opts, #{}),
             {ok, Watcher1} = aesc_fsm_min_depth_watcher:start_link(
-                               Type, TxHash, OnChainId, MinDepth, ?MODULE),
+                               Sub, TxHash, OnChainId, MinDepth, ?MODULE, WOpts),
             evt({watcher, Watcher1}),
             {ok, D1#data{watcher = Watcher1,
-                         latest = {watch, Type, TxHash, SignedTx, Updates}}};
+                         latest = {?MIN_DEPTH, Sub, TxHash, SignedTx, Updates}}};
+        {{?MIN_DEPTH, Sub}, Pid} when is_pid(Pid) ->
+            ok = aesc_fsm_min_depth_watcher:watch_for_min_depth(
+                   Pid, TxHash, MinDepth, ?MODULE, Sub),
+            {ok, D1#data{latest = {?MIN_DEPTH, Sub, TxHash, SignedTx, Updates}}};
+        {unlock, Pid} when Pid =/= undefined ->
+            ok = aesc_fsm_min_depth_watcher:watch_for_unlock(Pid, ?MODULE),
+            {ok, D1#data{latest = {watch, unlock, TxHash, SignedTx, Updates}}};
+        {close, Pid} when Pid =/= undefined ->
+            ok = aesc_fsm_min_depth_watcher:watch_for_channel_close(Pid, MinDepth, ?MODULE),
+            {ok, D1#data{latest = {watch, close, TxHash, SignedTx, Updates}}};
         {_, Pid} when Pid =/= undefined ->  % assertion
+            lager:debug("Unknown Type = ~p, Pid = ~p", [Type, Pid]),
```

Comment:
Shall this ever happen? Is this controlled by the remote party?

This sounds more a warning than a debug.